adapters/handlers/grpc/server.go

```diff
@@ -29,10 +30,24 @@ import (
 const maxMsgSize = 104858000 // 10mb, needs to be synchronized with clients
 
 func CreateGRPCServer(state *state.State) *GRPCServer {
-	s := grpc.NewServer(
+	o := []grpc.ServerOption{
 		grpc.MaxRecvMsgSize(maxMsgSize),
 		grpc.MaxSendMsgSize(maxMsgSize),
-	)
+	}
+
+	// Add TLS creds for the GRPC connection, if defined.
+	if len(state.ServerConfig.Config.GRPC.CertFile) > 0 && len(state.ServerConfig.Config.GRPC.KeyFile) > 0 {
```

Comment:
I think it would be a good idea to fail startup if only one of `GRPC.CertFile` or `GRPC.KeyFile` was provided. This prevents any unintentional running of an insecure server

Reply:
Agreed